exchanges/dispatch/dispatch.go

```diff
@@ -0,0 +1,307 @@
+package dispatch
+
+import (
+	"errors"
+	"fmt"
+
+	// "os"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/gofrs/uuid"
+)
+
+const (
+	// defaultGatewaySleep defines our sleep time in between worker generation
+	// to limit worker production
+	defaultGatewaySleep = time.Millisecond * 100
+
+	// defaultJobBuffer defines a maxiumum amount of jobs allowed in channel
+	// before we spill over and spawn more workers
+	defaultJobBuffer = 10
+
+	// defaultWorkerRate defines a rate at which we determine the efficient
+	// release of workers by dividing how many jobs are in the job queue/channel
+	// with this number
+	defaultWorkerRate = 4
+)
+
+// MaxWorkers define a ceiling for the amount of workers spawned
+var MaxWorkers int32 = 10
```

Comment:
Can link this up to main.go like https://github.com/thrasher-corp/gocryptotrader/blob/engine/main.go#L73